src/power/PowerComponentWin.cpp

```diff
@@ -13,3 +15,52 @@ void PowerComponentWin::doEnableScreensaver()
 {
   SetThreadExecutionState(ES_CONTINUOUS);
 }
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+bool PowerComponentWin::Suspend()
+{
+
+  return SetSuspendState(false, true, false) == TRUE;
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+bool PowerComponentWin::Reboot()
+{
+  if (!WinUtils::getPowerManagementPrivileges())
+  {
+    QLOG_ERROR() << "Failed to obtain power management, priviledges";
+    return false;
+  }
+
+  if (QSysInfo::WindowsVersion >= QSysInfo::WV_WINDOWS8)
+  {
+    return InitiateShutdownW(NULL, NULL, 0, SHUTDOWN_INSTALL_UPDATES | SHUTDOWN_RESTART,
+                             SHTDN_REASON_MAJOR_APPLICATION | SHTDN_REASON_MINOR_OTHER | SHTDN_REASON_FLAG_PLANNED) == ERROR_SUCCESS;
+  }
+  else
+  {
+    return InitiateShutdownW(NULL, NULL, 0, SHUTDOWN_INSTALL_UPDATES | SHUTDOWN_RESTART,
+                             SHTDN_REASON_MAJOR_APPLICATION | SHTDN_REASON_MINOR_OTHER | SHTDN_REASON_FLAG_PLANNED) == ERROR_SUCCESS;
+  }
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+bool PowerComponentWin::PowerOff()
+{
+  if (!WinUtils::getPowerManagementPrivileges())
+  {
+    QLOG_ERROR() << "Failed to obtain power management, priviledges";
+    return false;
+  }
+
+  if (QSysInfo::WindowsVersion >= QSysInfo::WV_WINDOWS8)
```

Comment:
Would it be shorter/less copy-paste-prone to set the flags based on platform, rather than the call?